src/main/java/io/deephaven/csv/containers/ByteSlice.java

```diff
@@ -74,7 +80,25 @@ public int size() {
         return end - begin;
     }
 
+    /** Gets the size of the slice. Same as {@link #size} */
+    @Override
+    public int length() { return size(); }
+
+    /** Gets the character at offset 'i' */
+    @Override
+    public char charAt(int i) {
+        return (char) data[begin + i];
+    }
+
+    @NotNull
+    @Override
+    public CharSequence subSequence(final int newBegin, final int newEnd) {
+        // Both newBegin and newEnd are relative to this.begin.
+        return new ByteSlice(data, begin + newBegin, begin + newEnd);
+    }
```

Comment:
We should likely do explicit bounds checking to adhere to the CharSequence throws contracts.

Reply:
charat needs it too!